File: Design_Pattern_and_Principles/FactoryMethodPatternExample/ConcreteDocuments.java
class WordDocument implements Document {
    public void open() {
        System.out.println("Opening Microsoft Word document...");
    }
}

class PdfDocument implements Document {
    public void open() {
        System.out.println("Opening Adobe PDF document...");
    }
}

class ExcelDocument implements Document {
    public void open() {
        System.out.println("Opening Microsoft Excel sheet...");
    }
}

File: Design_Pattern_and_Principles/FactoryMethodPatternExample/ConcreteDocuments.cs
using System;

class WordDocument : IDocument {
    public void Open() {
        Console.WriteLine("Opening Microsoft Word document...");
    }
}

class PdfDocument : IDocument {
    public void Open() {
        Console.WriteLine("Opening Adobe PDF document...");
    }
}

class ExcelDocument : IDocument {
    public void Open() {
        Console.WriteLine("Opening Microsoft Excel sheet...");
    }
}